python/lsst/ctrl/mpexec/cli/opt/input.py

```diff
@@ -0,0 +1,61 @@
+# This file is part of ctrl_mpexec.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+
+from lsst.daf.butler.cli.utils import MWOption, MWOptionDecorator, split_commas, unwrap
+
+
+class input_option(MWOptionDecorator):  # noqa: N801
+
+    defaultMetavar = "TASK[:LABEL]"
+
+    @staticmethod
+    def defaultHelp():
+        return unwrap("""Comma-separated names of the input collection(s). Entries may include a colon (:),
+                      the first string is a dataset type name that restricts the search in that
+                      collection.""")
+
+    @staticmethod
+    def optionFlags():
+        return ("--input",)
+
+    def __init__(self, callback=None, forward=False, help=None, metavar=defaultMetavar, multiple=False,
+                 required=False):
+        if callback is None:
+            self.callback = split_commas if multiple else None
+        else:
+            self.callback = callback
+        super().__init__(forward=forward)
+        self.help = help or self.defaultHelp()
+        self.metavar = metavar
+        self.multiple = multiple
+        self.required = required
+
+    def __call__(self, f):
+        return click.option(*self.optionFlags(), cls=MWOption,
+                            callback=self.callback,
+                            default=None if self.required else list(),
```

Comment:
just a general preference for immutable objects

Reply:
If you want immutable shouldn't it be a tuple?

Reply:
hah. Yes. 
> Is list() better than []
Trying again: I don't trust that passing `default=[]` won't end up sharing a list instance among multiple instances of the Option, and depending on how downstream code uses the default value chaos may ensue. `list()` is guaranteed to create a new list instance each time it is used and is safer. OTOH, `()` or `tuple()` would both be fine since `()` is immutable. Probably `()` is slightly better because as I understand it `tuple()` results in an extra line of code being executed (`return ()` or similar)

Reply:
and, click won't take a tuple as the default value of an option, so I'm going to stick with `list()`

Reply:
My question was mostly rhetorical, there is no real difference between `list()` and `[]`, both of them make a new empty list (but latter is shorter). What you pass as a parameter is not going to be shared, because it's a new list on every call. The sharing argument is only valid for the default argument value in the declaration of a method, there indeed you should not use mutable values as defaults.